proxy/proxy.go

```diff
@@ -55,6 +71,22 @@ func MiddlewareSetPrefixHeader() Middleware {
 	}
 }
 
+func MiddlewareSetTempoPrefixHeader() Middleware {
+	return func(r *http.Request) {
+		prefix, ok := getPrefix(r.Context())
+		if !ok {
+			return
+		}
+
+		// Do not override the prefix header if it is already set.
+		if r.Header.Get(PrefixHeader) != "" {
```

Comment:
Can this be a security issue?

So the gateway performs authn and authz on the user, and adds the correct `X-Scope-OrgID` header.
And with the above code, any successful authenticated user can impersonate any other user?

Reply:
Maybe I got confused with `PrefixHeader` vs. `TempoPrefixHeader` :confused:

Reply:
I fixed it, seems you're right.